### ISSUE TITLE
AlignArrayOfStructures crash on macro

### ISSUE THREAD

**Issue by 5cb1f72e-960e-48e1-90f9-6228c1f8eb4c** [llvm/llvm-project#51805]
|  |  |
| --- | --- |
| Bugzilla Link | [52463](https://llvm.org/bz52463) |
| Version | unspecified |
| OS | Windows NT |
| Attachments | [Stack backtrace](https://user-images.githubusercontent.com/95090403/143762702-12a78073-e98e-46bd-a358-1b6a2fd039f7.gz) |

## Extended Description 

Using the new AlignArrayOfStructures option with either "Left" or "Right" crashes on the following macro definition:

```
#define TEST                        \
	{                           \
		1,                  \
		{                   \
			1,          \
			1,          \
		},                  \
	},                          \

```
C:\Users\Jordan>clang-format --version
clang-format version 13.0.0

Attached is the stack backtrace.

**Comment by ambiso** [comment]
I think this is an alternative MWE

```cpp
int arr[2][1] = {
    {1,}, {1,},
};
```

or also:

```cpp
#include <vector>

std::vector<int> arr[] = {
    {1,}, {1,},
};
```

**Comment by mydeveloperday** [comment]
I think this is fixed in the main branch.. none of these examples crash, let me pull out the commit that solved this and add it to the issue

**Comment by mydeveloperday** [comment]
I think its fixed by https://reviews.llvm.org/D110481 landed in commit a36227cb2b6a14fc30bfd303328d1d1abb632010
